pyqtgraph/parametertree/parameterTypes/checklist.py

```diff
@@ -0,0 +1,169 @@
+
+    def updateLimits(self, _param, limits):
+        oldOpts = self.names
+        val = self.opts['value']
+        # Make sure adding and removing children don't cause tree state changes
+        self.blockTreeChangeSignal()
+        self.clearChildren()
+        self.forward, self.reverse = ListParameter.mapping(limits)
+        for chName in self.forward:
+            # Recycle old values if they match the new limits
+            newVal = bool(oldOpts.get(chName, False))
+            child = self.create(type='bool', name=chName, value=newVal, default=None)
+            self.addChild(child)
+            # Prevent child from broadcasting tree state changes, since this is handled by self
+            child.blockTreeChangeSignal()
+            child.sigValueChanged.connect(self._onSubParamChange)
+        # Purge child changes before unblocking
+        self.treeStateChanges.clear()
+        self.unblockTreeChangeSignal()
+        self.setValue(val)
+
+    def _onSubParamChange(self, param, value):
+        # Old value will become false, new value becomes true. Only fire on new value = true for exclusive
+        if not value and self.opts['exclusive']:
+            # Re-allow selection
+            return
+        elif self.opts['exclusive']:
+            val = self.reverse[0][self.reverse[1].index(param.name())]
+            return self.setValue(val)
+        # Interpret value, fire sigValueChanged
+        return self.setValue(self.value())
+
+    def optsChanged(self, param, opts):
+        if 'exclusive' in opts:
+            # Force set value to ensure updates
+            # self.opts['value'] = self._VALUE_UNSET
+            self.setValue(self.opts['value'])
+
+    def value(self):
+        vals = [self.forward[p.name()] for p in self.children() if p.value()]
+        exclusive = self.opts['exclusive']
+        if not vals and exclusive:
+            return None
+        elif exclusive:
+            return vals[0]
+        else:
+            return vals
+
+    def setValue(self, value, blockSignal=None):
+        self.targetValue = value
+        exclusive = self.opts['exclusive']
+        # Will emit at the end, so no problem discarding existing changes
+        cmpVals = value if isinstance(value, list) else [value]
+        for ii in range(len(cmpVals)-1, -1, -1):
+            exists = any(fn.eq(cmpVals[ii], lim) for lim in self.reverse[0])
+            if not exists:
+                del cmpVals[ii]
+        names = [self.reverse[1][self.reverse[0].index(val)] for val in cmpVals]
+        if exclusive and len(names) > 1:
+            names = [names[0]]
+        elif exclusive and not len(names) and len(self.forward):
+            # An option is required during exclusivity
+            names = [self.reverse[1][0]]
+        for chParam in self:
+            checked = chParam.name() in names
+            chParam.setValue(checked, self._onSubParamChange)
+            chParam.setOpts(enabled=not (exclusive and checked))
```

Comment:
Commenting out this line fixed the issue with the checkboxes looking disabled as they're being selected when `exclusive` is `True`.

Reply:
The disabling was by design, the goal being when `exclusive` is true exactly one option must be selected (like `list`). This keeps the currently checked box from becoming unchecked

Reply:
@j9ac9k Do you think it would be better to have custom logic that leaves the checkbox looking enabled, while also disallowing the user from having a `None` selection? I'm open to this, but I think leaving it disabled is a clearer indicator that something must be selected

Reply:
Sorry been slow to respond, I'm a bit under the weather as is one of my kids.  

_Generally_ speaking, when you can only select one entry, you don't have checkboxes, you get [radio buttons](https://doc.qt.io/qt-5/qradiobutton.html).  The exclusive with check-boxes feel a little out of place; but without knowing how much code would have to change to accommodate the radio buttons vs. the checkbox; I'm ok with it.

That said, having the checkboxes appear disabled, and yet be selectable definitely feels like an odd combination.

I guess I'll turn this around a bit, how difficult would it be to swap out checkboxes for QRadioButton's if `'exclusive': True` is passed?

Reply:
Probably not that hard -- I am already making a button group in the parameter item, so I'll play with it a bit